lib/maths/CTimeSeriesDecompositionDetail.cc

```diff
@@ -599,10 +599,22 @@ void CTimeSeriesDecompositionDetail::CPeriodicityTest::test(const SAddValue& mes
     }
 }
 
-void CTimeSeriesDecompositionDetail::CPeriodicityTest::clear(ETest test, core_t::TTime time) {
-    if (m_Windows[test] != nullptr) {
-        m_Windows[test].reset(this->newWindow(test));
-        m_Windows[test]->initialize(time);
+void CTimeSeriesDecompositionDetail::CPeriodicityTest::maybeClear(core_t::TTime time,
+                                                                  double shift) {
+    for (auto& test : {E_Short, E_Long}) {
```

Comment:
Could it just be `auto` for iterating simple enum values?

Reply:
This was an oversight from originally iterating over the windows rather than the enums